lib/puppet/indirector/hiera.rb

```diff
@@ -18,6 +18,7 @@ def initialize(*args)
   def find(request)
     not_found = Object.new
     options = request.options
+    Puppet.debug "Performing a hiera indirector lookup of #{request.key} with options #{options.inspect}"
     value = hiera.lookup(request.key, not_found, Hiera::Scope.new(options[:variables]), nil, convert_merge(options[:merge]))
```

Comment:
This call should not go to hiera! It should go to the Puppet::Pops::Lookup (implementation of the lookup class). That way it works with Data in Modules as well.